server/src/main/java/org/elasticsearch/features/FeaturesSupportedSpecification.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.features;
+
+import java.util.Set;
+
+public class FeaturesSupportedSpecification implements FeatureSpecification {
+    @Override
+    public Set<NodeFeature> getFeatures() {
+        // a feature that indicates cluster features are supported
+        return Set.of(new NodeFeature("features_supported"));
```

Comment:
This is a placeholder for all "core" (non module/plugins) supported features, right?
Can you add a javadoc here?

Reply:
It's not really a placeholder, its a feature to indicate a node supports features. Nodes that don't support features won't have this - it's a way to check feature support without checking the version

Reply:
Oh, clever! (anyway, the comment about javadoc stands - something like "The provider (specification?) for features supported by server core" or something similar would be useful)

Reply:
Good idea